wallet/wallet.go

```diff
@@ -261,6 +261,10 @@ func (w *Wallet) WaitForShutdown() {
 	}
 	w.chainClientLock.Unlock()
 	w.wg.Wait()
+
+	//We can close the db now that we know all go routines have exited and
+	//we are sure no further attempts to write to the wallet db will be done.
+	w.db.Close()
```

Comment:
Thanks for your review @halseth and sorry for my late response, somehow lost track of this issue.
To see that I understand: 
1. I will leave the onClientConnected function since it adds a waiting for the syncToChain go routine that is needed.
2. I will remove the db.Close call.

The impact of this is that users that uses btcwallet as library will be responsible to call loader.UnloadWallet on shutdown and not only wallet.Stop (as it is now in LND).
If this is what you think needed I will need to create another fix in LND to explicitly call UnloadWallet on shutdown.
LMK your thoughts.

Reply:
At the moment I removed the db.Close and rebased.